frontend/src/metabase/modes/components/drill/ColumnFilterDrill.jsx

```diff
@@ -23,10 +23,17 @@ export default function ColumnFilterDrill({ question, clicked }) {
   }
 
   const { column } = clicked;
-  const initialFilter = new Filter([], null, query).setDimension(
-    column.field_ref,
-    { useDefaultOperator: true },
-  );
+
+  // if our field ref doesn't match our column id, we have a remapped column
+  // and need to use the field id instead of the ref
+  const fieldRef =
+    column.id !== undefined && column.id !== column.field_ref[1]
+      ? ["field", column.id, null]
```

Comment:
should we copy over the third element from the `column.field_ref` here? I'm not quite sure what they're for though.

Reply:
🤷  I'm not sure what it does either. I don't think i've seen it populated.

Reply:
There are some places where it's populated, but I'm not totally sure what they're used for or have a use for anything here specifically. e.g.
- https://github.com/metabase/metabase/blob/a0d5079de8315e21dc297e38c95ef1b8c2542419/frontend/test/metabase/lib/dataset.unit.spec.js#L154
- https://github.com/metabase/metabase/blob/a0d5079de8315e21dc297e38c95ef1b8c2542419/frontend/src/metabase/parameters/utils/mapping-options.unit.spec.js#L90
- https://github.com/metabase/metabase/blob/a0d5079de8315e21dc297e38c95ef1b8c2542419/frontend/test/metabase/lib/dataset.unit.spec.js#L233
- https://github.com/metabase/metabase/blob/a0d5079de8315e21dc297e38c95ef1b8c2542419/frontend/test/metabase/lib/query_time.unit.spec.js#L34

for me `null` indicate that the 3rd element is something this code won't use *for sure*. While copying it over is always look safer from the previous assumption, it would create noise if the 3rd element *isn't actually used* here. Now I'm on the fence about this since we both don't know what it's used for and whether it's being used here. Maybe other engineers would know @ranquild do you know what the 3rd item is used for and if it's used here? Or could you help tag other people who might know the answer. Thanks!

Reply:
The 3rd param is dimension options. `["field", 100, { "temporal-unit": "month" }]` - this is subdimension of `["field", 100, null]`. So passing `null` would likely change the behavior of this code. @WiNloSt @iethree  Could you test this with columns from aggregated results?

Reply:
When adding aggregation, the column mapping disappears. Not sure if that's supposed to happen.

What I did is using the repro steps to create "M1" and then from there count by category + created (by year) after saving I'll get a new question then after turning that into a model, there won't be an option to map which result column correspond to which table column.

I assume what this tries to fix is for a model created from SQL question, which I'm not sure there would be a subdimension in such a case.

@nemanjaglumac Is it possible to create a repo that the subdimension (3rd item in the MBQL field) isn't null?

Reply:
In my testing, copying the field options breaks filters altogether for aggregated queries. The filters are simply invalid, and exhibit the same behavior as listed in the issue (you can't save them).

Mapping an aggregated field to a database field in a model makes any filter on that aggregated field throw an error. I think this is nonsensical, but I could see how a user would do it for something like a `max()` aggregation. Because this is a native query, I don't think we can detect if this is an aggregate column to disable filters on that field?

![Screen Shot 2022-08-08 at 4 05 32 PM](https://user-images.githubusercontent.com/30528226/183522759-ed20e7c2-8cfc-4009-b78b-941ea18aca55.png)

Notably, filters from notebook mode and the bulk filter modal, have the same limitation.

In any event, I think we should merge this as-is. We definitely have some lingering limitations here with SQL-based models with aggregations, but this brings the column header filters to parity with the notebook and modal filters (i.e. they work for all non-aggregated fields when remapped).

Reply:
That's fair, I had some troubles with mapping column + aggregation when copying over the subdimension part as well. Seems reasonable to merge this as-is given the circumstances.

Reply:
> @nemanjaglumac Is it possible to create a repo that the subdimension (3rd item in the MBQL field) is being used?

@WiNloSt not sure what **exactly** you had in mind, but asserting on that field would be possible in a repro, yes.